dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/RelationshipRestRepository.java

```diff
@@ -60,4 +90,101 @@ public RelationshipRest findOne(Context context, Integer integer) {
     public DSpaceResource<RelationshipRest> wrapResource(RelationshipRest model, String... rels) {
         return new RelationshipResource(model, utils, rels);
     }
+
+    @Override
+    protected RelationshipRest createAndReturn(Context context, List<DSpaceObject> list)
+        throws AuthorizeException, SQLException, RepositoryMethodNotImplementedException {
+
+        HttpServletRequest req = getRequestService().getCurrentRequest().getHttpServletRequest();
+
+        Relationship relationship = new Relationship();
+        if (list.size() == 2 && list.get(0).getType() == Constants.ITEM && list.get(1).getType() == Constants.ITEM) {
+            Item leftItem = (Item) list.get(0);
+            Item rightItem = (Item) list.get(1);
+            RelationshipType relationshipType = relationshipTypeService
+                .find(context, Integer.parseInt(req.getParameter("relationshipType")));
+
+            relationship.setLeftItem(leftItem);
+            relationship.setRightItem(rightItem);
+            relationship.setRelationshipType(relationshipType);
+            try {
+                relationship = relationshipService.create(context, relationship);
+            } catch (AuthorizeException e) {
+                throw new AccessDeniedException("You do not have write rights on this relationship's items");
+            }
+            return relationshipConverter.fromModel(relationship);
+        } else {
+            throw new UnprocessableEntityException("The given items in the request were not valid items");
+        }
+
+
+    }
+
+    @Override
+    protected RelationshipRest put(Context context, HttpServletRequest request, String apiCategory, String model,
+                                   Integer id, List<DSpaceObject> dSpaceObjects)
+        throws RepositoryMethodNotImplementedException, SQLException, AuthorizeException {
+
+        Relationship relationship = relationshipService.find(context, id);
+        if (relationship == null) {
+            throw new ResourceNotFoundException(apiCategory + "." + model + " with id: " + id + " not found");
+        }
+
+        if (dSpaceObjects.size() == 2 && dSpaceObjects.get(0).getType() == Constants.ITEM
+            && dSpaceObjects.get(1).getType() == Constants.ITEM) {
+            Item leftItem = (Item) dSpaceObjects.get(0);
+            Item rightItem = (Item) dSpaceObjects.get(1);
+
+            if (isAllowedToModifyRelationship(context, relationship, leftItem, rightItem)) {
```

Comment:
This permissions check seems to duplicate what is already in the `relationshipService.update()` call.  Is there a reason we aren't just catching the `AuthorizeException` and turning it into an` AccessDeniedException`, like in `createAndReturn` above?

Reply:
We'll verify this in detail, I would expect this can indeed be adjusted

Reply:
Raf has verified this in detail, and the `relationshipService.update()` call is unable to verify the same permissions as the `isAllowedToModifyRelationship` function here.
The `relationshipService.update()` is only aware of the new items, and no longer references the current items.
The `isAllowedToModifyRelationship` method also verifies the user is authorized to update the current items: https://github.com/atmire/DSpace/blob/w2p-57159_permission-to-create-relations/dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/RelationshipRestRepository.java#L165

Reply:
Thanks for checking. I guess we can just leave this as-is for now.